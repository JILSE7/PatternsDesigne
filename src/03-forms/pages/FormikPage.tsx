import React from 'react'
import '../styles/styles.css'

import {FormikErrors, useFormik} from 'formik'


interface IFORMVALUES {
    firstName: string,
    lastName: string,
    email: string
}

const FormikPage = () => {

    const validate = ({email,lastName,firstName}:IFORMVALUES) => {

        const errors:FormikErrors<IFORMVALUES> = {};

        
        if(!firstName.trim())errors.firstName = 'Required'
        else if(firstName.length > 15)errors.firstName = 'FirstName must be max 15 characters'

        if(!lastName.trim())errors.lastName = 'Required'
        else if(lastName.length > 10)errors.lastName = 'LastName must be max 10 characters'

        if (!values.email) {
            errors.email = 'Required';
        } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email)) {
            errors.email = 'Invalid email address';
        }

        return errors;

    }

    const {handleSubmit,handleChange, values, errors, touched, handleBlur} = useFormik({
        initialValues:{
            firstName: '',
            lastName: '',
            email: ''
        }, 
        onSubmit: (values,e) => {
            console.log(values);    
        },
        validate
    });

    


  return (
    <div>
        <h1>Formik Tutorial</h1>

        <form onSubmit={handleSubmit}>
            <label htmlFor="firstName" >FirstName</label>
            <input type="text" name='firstName' id='firstName' value={values.firstName} onChange={handleChange} onBlur={handleBlur}/>
            {touched.firstName && errors.firstName && <span>{'First Name is ' + errors.firstName}</span>}

            <label htmlFor="lastName">LastName</label>
            <input type="text" name='lastName' id='lastName' value={values.lastName} onChange={handleChange} onBlur={handleBlur}/>
            {touched.lastName && errors.lastName && <span>{'Last Name is ' + errors.lastName}</span>}

            <label htmlFor="email" >Email</label>
            <input type="text" name='email' id='email' value={values.email} onChange={handleChange} onBlur={handleBlur}/>
            {touched.email && errors.email && <span>{errors.email}</span>}
            
            

            <button type='submit' >Iniciar Sesi√≥n</button>
        </form>
    </div>
  )
}

export default FormikPage