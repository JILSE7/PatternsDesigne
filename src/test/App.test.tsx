import { render } from '@testing-library/react';
import React from 'react';
//import * as ReactDOM from 'react-dom';
import TestRenderer, { create,act } from 'react-test-renderer';


import FormikPage from '../03-forms/pages/FormikPage';




console.log(create);

describe('Pruebas al componente Raiz', () => {
    
    test('Deebe mostrar algo', () => { 
        const wrapper = create(<FormikPage/>);
        console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot()
     });


     test('Deebe mostrar algo', () => { 
        const {container} = render(<FormikPage/>);

        console.log(container);
     });



})