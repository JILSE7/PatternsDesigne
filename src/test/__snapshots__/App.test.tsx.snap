// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente Raiz Deebe mostrar algo 1`] = `
<div>
  <h1>
    Formik Tutorial
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="firstName"
    >
      FirstName
    </label>
    <input
      id="firstName"
      name="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="lastName"
    >
      LastName
    </label>
    <input
      id="lastName"
      name="lastName"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Iniciar Sesi√≥n
    </button>
  </form>
</div>
`;
